tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -131,31 +144,132 @@ func (a *AwsProvider) CreateWindowsVM() (rerr error) {
+		return nil, err
+	}
+	// Get the public IP
+	publicIPAddress, err := a.GetPublicIP(instanceID)
 	if err != nil {
-		return fmt.Errorf("failed to record instance ID to file at '%s',instance will not be able to be deleted, "+
+		return nil, err
+	}
+	// Build new credentials structure to be used by other actors
+	credentials = types.NewCredentials(instanceID, publicIPAddress, decryptedPassword)
+	err = resource.AppendInstallerInfo([]string{instanceID}, []string{}, a.resourceTrackerDir)
+	if err != nil || credentials == nil {
+		return nil, fmt.Errorf("failed to record instance ID to file at '%s',instance will not be able to be deleted, "+
 			"%v", a.resourceTrackerDir, err)
 	}
 
 	// Output commandline message to help RDP into the created instance.
-	log.V(0).Info(fmt.Sprintf("Successfully created windows instance: %s, please RDP into the Windows instance.",
-		instanceID))
+	log.V(0).Info(fmt.Sprintf("Successfully created windows instance: %s, "+
+		"please RDP into the Windows instance created at %s using Admininstrator as user and %s password",
+		instanceID, publicIPAddress, decryptedPassword))
+	// TODO: Output the information to a file
+	return credentials, nil
+}
 
-	return nil
+// GetPublicIP returns the public IP address associated with the instance. Make to sure to call this function
+// after the instance is in running state. Exposing this function to be used in testing later.
+func (a *AwsProvider) GetPublicIP(instanceID string) (string, error) {
+	// Till the instance comes to running state we cannot get the public ip address associated with it.
+	// So, it's better to query the AWS api again to get the instance object and the ip address.
+	instance, err := a.GetInstance(instanceID)
+	if err != nil {
+		return "", err
+	}
+	return *instance.PublicIpAddress, nil
+}
+
+// GetPassword returns the password associated with the string. Exposing this to be used in tests later
+func (a *AwsProvider) GetPassword(instanceID string) (string, error) {
+	privateKey, err := ioutil.ReadFile(a.privateKeyPath)
+	if err != nil {
+		return "", err
+	}
+	privateKeyBytes := privateKey
+	decodedEncryptedData, err := a.getDecodedPassword(instanceID)
+	if err != nil {
+		return "", err
+	}
+	decryptedPwd, err := rsaDecrypt(decodedEncryptedData, privateKeyBytes)
+	if err != nil {
+		return "", err
+	}
+	return string(decryptedPwd), nil
+}
+
+// getDecodedPassword gets the decoded password from the AWS cloud provider API
+func (a *AwsProvider) getDecodedPassword(instanceID string) ([]byte, error) {
+	// The docs within the aws-sdk says
+	// `If you try to retrieve the password before it's available,
+	// 	the output returns an empty string. We recommend that you wait up to 15 minutes
+	// after launching an instance before trying to retrieve the generated password.`
+	// So let's have hard timeout of 15 minutes.
+	startTime := time.Now()
+	var decodedPassword []byte
+	for {
+		currTime := time.Since(startTime)
+		if currTime >= 15*time.Minute {
```

Comment:
is there any reason we are not using the aws WaitUntilPasswordDataAvailable function ? https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.WaitUntilPasswordDataAvailable

Reply:
You won't get the password needed, if you look at the code of the function, it just has a wait loop with no extension points.

Reply:
But if your point is for the sake of consistency, I am fine with changing it. Let me know, if you think that is the case

Reply:
For future notice, there is a way to specify the password https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/ec2launch.html per our discussion about matching with Azure letting you set the password.

Reply:
Yeah, it's not setting the password per say, it like creating a golden image with the password you want.

Reply:
>You won't get the password needed, if you look at the code of the function, it just has a wait loop with no extension points.

I don't see why we can't use `EC2.WaitUntilPasswordDataAvailable()` and then get the password. This will remove the need for the loop and will make the code look much cleaner.

If you don't want to do this, at least make 15 minutes a constant.